lalrpop/src/grammar/parse_tree.rs

```diff
@@ -44,12 +44,54 @@ impl Into<Box<Content>> for Span {
 
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub enum GrammarItem {
+    MatchToken(MatchToken),
     ExternToken(ExternToken),
     InternToken(InternToken),
     Nonterminal(NonterminalData),
     Use(String),
 }
 
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct MatchToken {
+    pub contents: Vec<MatchContents>
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct MatchContents {
+    pub items: Vec<MatchItem>
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum MatchItem {
+    Unmapped(MatchSymbol, Span),
+    Mapped(MatchSymbol, MatchMapping, Span)
```

Comment:
I suspect that `CatchAll` could be here. That is, I don't think that `_ => "foo"` makes any sense.